### PR TITLE
allow mobx to run in web worker

### PR DESCRIPTION
**Pull request by shahata**
currently `getGlobal` is first invoked from an iife which initializes `globalState`, which means `mockGlobal` is not yet initialized.

Thanks for taking the effort to create a PR!

If you are creating an extensive PR, you might want to open an issue with your idea first, so that you don't put a lot of effort in an PR that wouldn't be accepted. Please prepend pull requests with `WIP: ` if they are not yet finished
PR checklist:

* [ ] Added unit tests
* [ ] Updated changelog
* [ ] Updated `/docs`. For new functionality, at least `API.md` should be updated
* [ ] Added typescript typings
* [ ] Verified that there is no significant performance drop (`npm run perf`)

Feel free to ask help with any of these boxes!

The above process doesn't apply to doc updates etc.

### CODE REVIEW

File: src/core/globalstate.ts [hunk 1]
```diff
@@ -143,6 +143,7 @@ export class MobXGlobals {
 
 let canMergeGlobalState = true
 let isolateCalled = false
+const mockGlobal = {}
```
Comment (FredyC) [inline src/core/globalstate.ts]: Why did you move it here? Let's keep it closer to the actual use.

Comment (shahata) [reply src/core/globalstate.ts]: There's a good reason to move it. I described the issue in the commit message. Currently when `getGlobal` is first invoked, it throws exception in case `mockGlobal` is returned because `mockGlobal` is not initialized yet.

Comment (shahata) [reply src/core/globalstate.ts]: I can move the whole function upward in order to keep them closer if you prefer

Comment (shahata) [reply src/core/globalstate.ts]: Just to make it clearer. This is in general the current code:

```js
export let globalState = (function() {
    const global = getGlobal()
    ... global._mobxGlobals ...
})()

const mockGlobal = {}
export function getGlobal() {
  ...
  return mockGlobal
}
```

since `globalState` is a IIFE which is invoked before the line `const mockGlobal = {}` is reached, the `mockGlobal` variable is still not initialized  with the `{}` value, therefore the call to `getGlobal` inside the IIFE will not work as expected.

Comment (FredyC) [reply src/core/globalstate.ts]: Ah, you are right, missed it's being called in IIFE. Yea, I think moving `getGlobal` on top as well makes sense to keep it collocated.

**Comment by FredyC** [comment]
Also please, make another PR for `mobx4-master`. A cherry-pick should be fine probably.

File: src/core/globalstate.ts (hunk 1 above)
Comment (shahata) [reply src/core/globalstate.ts]: done

**Comment by shahata** [comment]
@FredyC I think it is ready.
1) hoisted the whole function to be before the IIFE
2) added changelog
3) created #2185 for applying changes also to `mobx4-master`

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+-   Allow mobx to run in web worker
```
Comment (FredyC) [inline CHANGELOG.md]:
```suggestion
-   Fix running mobx in web worker [#2184](https://github.com/mobxjs/mobx/pull/2184/files)
```

**Comment by FredyC** [comment]
Thanks for contribution

**Comment by shahata** [comment]
@FredyC any estimation when to expect next version to be published?

**Comment by mweststrate** [comment]
Published as 5.15.0
